### PR TITLE
Fix regenerating site for change in excluded files while tasting

### LINKED ISSUE #264: Taste trigers complete site regeneration for excluded items
While using VisualStudio Code to write posts and layouts in a git version controlled site source I have noticed that while tasting the site was continously being regenerated for changes in the .git folder.
I have added the folder and git items to the config as follows:
~~~yaml
exclude:
  - .gitignore
  - .git
  - taste.bat
  - Pretzel.exe
~~~
Now the excluded items did not get copied, but were still triggering the regeneration.

### PR DESCRIPTION
**Pull request by kodfodrasz**
Taste mode does not trigger complete regeneration of the change of a file excluded in the configuration.

Fixes #264

### CODE REVIEW

**Comment by kodfodrasz** [comment]
It was a bit too early to submit this PR. The first attempt for the fix was not correct :(

File: src/Pretzel.Logic/Templating/Context/SiteContextGenerator.cs [hunk 1]
```diff
@@ -186,12 +186,12 @@ private bool ContainsYamlFrontMatter(string file)
...
         }
 
-        private bool IsExcludedPath(string relativePath)
+        public bool IsExcludedPath(string relativePath)
         {
             return excludes.Contains(relativePath) || excludes.Any(e => relativePath.StartsWith(e));
         }
 
-        private bool IsIncludedPath(string relativePath)
+        public bool IsIncludedPath(string relativePath)
```
Comment (laedit) [inline src/Pretzel.Logic/Templating/Context/SiteContextGenerator.cs]: This is not necessary.

File: src/Pretzel.Logic/Templating/Context/SiteContextGenerator.cs [hunk 2]
```diff
@@ -186,12 +186,12 @@ private bool ContainsYamlFrontMatter(string file)
             return postFirstLine != null && postFirstLine.StartsWith("---");
         }
 
-        private bool IsExcludedPath(string relativePath)
+        public bool IsExcludedPath(string relativePath)
```
Comment (laedit) [inline src/Pretzel.Logic/Templating/Context/SiteContextGenerator.cs]: If it's public you have to make sure that there is the corresponding unit tests.

**Comment by laedit** [comment]
Thanks for your PR!
I just have some concerns, see the comments on the commit.

File: src/Pretzel.Logic/Templating/Context/SiteContextGenerator.cs (hunk 2 above)
Comment (kodfodrasz) [inline src/Pretzel.Logic/Templating/Context/SiteContextGenerator.cs]: Ok, will add unit tests.

File: src/Pretzel.Logic/Templating/Context/SiteContextGenerator.cs (hunk 1 above)
Comment (kodfodrasz) [inline src/Pretzel.Logic/Templating/Context/SiteContextGenerator.cs]: True, I only changed visibility to be "semantically symmetric" with IsExcludedPath

Comment (laedit) [inline src/Pretzel.Logic/Templating/Context/SiteContextGenerator.cs]: You have a point about the "semantically symmatric", leave the change and just add some unit tests please.

**Comment by thoemmi** [comment]
I was just about to create a similar pull request when I sumbled over this one :smile: 
@kodfodrasz Are you going to add some unit tests? Otherwise I'll create a new complete PR.

**Comment by kodfodrasz** [comment]
Hi!
Hang on, I'll try to get to finish this one on this weekend. It has been delayed slightly since I submitted it :) Now that there is someone interested in it i'll prioritize this.
Date: Wed, 18 Nov 2015 23:55:09 -0800

**Comment by kodfodrasz** [comment]
I hope these tests are enough. I don't think these methods need anything more complicated.

**Comment by laedit** [comment]
Thanks! :smiley:

**Comment by thoemmi** [comment]
:+1: Thanks

**Pull request merged**